scripts/install/autobrr.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+# autobrr installer
+# ludviglundgren 2021 for Swizzin
+
+#shellcheck source=sources/functions/utils
+. /etc/swizzin/sources/functions/utils
+
+app_reqs=("curl" "sqlite3")
+apt_install "${app_reqs[@]}"
+
+users=($(_get_user_list))
+
+if [[ -n $1 ]]; then
+    user=$1
+    _autobrr_user_config ${user}
+    if [[ -f /install/.nginx.lock ]]; then
+        echo_progress_start "Configuring nginx"
+        bash /etc/swizzin/scripts/nginx/autobrr.sh
+        systemctl reload nginx
+        echo_progress_done
+    fi
+    exit 0
+fi
+
+_install_autobrr() {
+
+    echo_progress_start "Downloading release archive"
+
+    case "$(_os_arch)" in
+        "amd64") arch='x86_64' ;;
+        "arm64") arch="arm64" ;;
+        "armhf") arch="armv6" ;;
+        *)
+            echo_error "Arch not supported"
+            exit 1
+            ;;
+    esac
+
+    latest=$(curl -sL https://api.github.com/repos/autobrr/autobrr/releases/latest | grep "linux_$arch" | grep browser_download_url | cut -d \" -f4) || {
+        echo_error "Failed to query GitHub for latest version"
+        exit 1
+    }
+
+    if ! curl "$latest" -L -o "/tmp/autobrr.tar.gz" >> "$log" 2>&1; then
+        echo_error "Download failed, exiting"
+        exit 1
+    fi
+    echo_progress_done "Archive downloaded"
+
+    echo_progress_start "Extracting archive"
+
+    # the archive contains both autobrr and autobrrctl to easily setup the user
+    tar xfv "/tmp/autobrr.tar.gz" --directory /usr/bin/ >> "$log" 2>&1 || {
+        echo_error "Failed to extract"
+        exit 1
+    }
+    rm -rf "/tmp/autobrr.tar.gz"
+    echo_progress_done "Archive extracted"
+}
+
+_autobrr_user_config() {
+    echo_progress_start "Configuring autobrr"
+
+    # get random available port
+    port=$(port 10000 12000)
+
+    # generate a sessionSecret
+    sessionSecret="$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c16)"
+
+    if [ ! -d "/home/$user/.config/autobrr/" ]; then
+        mkdir -p "/home/$user/.config/autobrr/"
+    fi
+    chown -R "$user": "/home/$user/.config/autobrr"
+
+    cat > "/home/$user/.config/autobrr/config.toml" << CFG
+# config.toml
+# Hostname / IP
+#
+# Default: "localhost"
+#
+host = "127.0.0.1"
+# Port
+#
+# Default: 8989
+#
+port = ${port}
+# Base url
+# Set custom baseUrl eg /autobrr/ to serve in subdirectory.
+# Not needed for subdomain, or by accessing with the :port directly.
+#
+# Optional
+#
+baseUrl = "/autobrr/"
+# autobrr logs file
+# If not defined, logs to stdout
+#
+# Optional
+#
+#logPath = "log/autobrr.log"
```

Comment:
This would be nice to enable, `~/.config/autobrr/log.txt` or similar would be good enough. Especially if the output is set to `DEBUG`, that could fill up the `journalctl` fast. Had that happen with transmission before (spamming stdout while in service) and it just _kills_ performance of all `systemctl` etc. commands

Reply:
Interesting! Currently it outputs to both the console and logfile if set. It has mechanisms to rotate logs as well so it doesn't fill up the disk. Will have to look into writing only to log file if set, and nothing to console then.

Reply:
I set it to log to log to `/home/$user/.config/autobrr/logs/autobrr.log` and will make some changes to only have some stuff output to console/stdout. But that will come with an update to the app :)